test/integration/aggregates/aggregates_integration_test.go

```diff
@@ -168,48 +165,49 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
 		series := set.At()
 		lset := series.Labels()
 		aggr := lset.Get("Aggregator")
-		fmt.Println("\n\nLables:", lset)
+		//fmt.Printf("\nLables: %v", lset)
+
 		iter := series.Iterator()
 		for iter.Next() {
 
-			tm, v := iter.At()
-			fmt.Printf("t=%d,v=%f; ", tm, v)
-			if aggr == "count" {
-				totalCount += int(v)
-			}
-
-			if tm > testConfig.testStartTime && tm < testConfig.testEndTime-testConfig.queryStep {
-
-				assert.Condition(t, func() bool { return aggr == "count" && int64(v) != expectedCount },
-					"Count is not ok - expected %d, got %f at time %d", expectedCount, v, tm)
-
-				assert.Condition(t, func() bool { return aggr == "sum" && v != expectedSum },
-					"Sum is not ok - expected %f, got %f at time %d", expectedSum, v, tm)
-
-				assert.Condition(t, func() bool { return aggr == "avg" && v != expectedSum/float64(expectedCount) },
-					"Avg is not ok - expected %f, got %f at time %d", expectedSum/float64(expectedCount), v, tm)
+			_, v := iter.At()
+			//fmt.Printf("t=%d,v=%f\n", tm, v)
+
+			switch aggr {
+			case "count":
+				actualCount += int(v)
+			case "sum":
+				actualSum += v
+			case "avg":
+				if ! math.IsNaN(v) {
+					actualAvgResultsCount += 1
+					actualAvgResultsSum = actualAvgResultsSum + v
```

Comment:
Not sure i understand what you count here. is it because there is an AVG per step?